### DISCUSSION TITLE
React multi select work with Tags and groups in custom objects

### DISCUSSION THREAD

**Discussion by yaman3bd** [floating-ui/floating-ui#1928] [Q&A]
hello, I am trying to implement Tags select so that the user could select more than one tag and my options are groups like:
```js
options: [
    {
      label: 'DC',
      options: [
        {label: 'Batman', value: 'Batman'},
        {label: 'Robin', value: 'Robin'},
        {label: 'Superman', value: 'Superman'}]
    },
    {
      label: 'Marvel',
      options: [
        {label: 'Spiderman', value: 'Spiderman'},
        {label: 'Iron Man', value: 'Iron Man'},
        {label: 'Captain America', value: 'Captain America'}
      ],
    },
  ]
```
but I am not sure how can I implement this, I have looked into the examples but I am having some problems when I want to change the selected options style and how to handle the groups when the user navigates using arrows,
I have only the `activeIndex` which only has a value of `6` the count of each group options
any help implementing this?
and maybe we can add it to the provided examples?

**Comment by atomiks** [comment]
Multi-select is definitely missing from the examples, but I believe it's possible to implement with the current version. `selectedIndex` does only allow one value, not an array, but I think you can still work with it.

Do you have a CodeSandbox demo of what you've tried so far?

**Reply by yaman3bd** [reply]
unfortunately, I could not get `CodeSandbox` to work with the example I made,
because I am using `Vite`, `tailwindcss` and some other tools

however, I made an example repe
[https://github.com/yaman3bd/floating-ui-select-example](https://github.com/yaman3bd/floating-ui-select-example
)
please pull the code and run the project,
but the problems I have:
when I have grouped options and the user uses the keyboard arrows and hit Enter I get error because `activeIndex` is not the index for the current option, the problem is because my options object is like:
```js
options: [
    {
      label: 'names',
      options: [
        {value: 'yaman3bd', label: 'Yaman'},
        {value: 'hussam3bd', label: 'Hussam'},
        {value: 'ali3bd', label: 'Ali'}
      ]
    },
    {
      label: 'Jobs',
      options: [
        {value: 'react', label: 'Frontend'},
        {value: 'laravel', label: 'Backend'},
        {value: 'react, laravel', label: 'Fullstack'}
      ]
    },
  ];
  ```
so in this case I need the group index and the option index in the group for example:
If I have 2 groups one has 3 options and the other has 5 options
and the user now is on the 2nd group's last option 
so active index should be: `4`
not `7`
means that active index should be based on the current group options count
not total options
I could not get it to work the same thing I wanted
and for the tags, I used an array of selected options and I did:
```js
function handleSelect() {
    setSelectedIndex(index ?? 0);
    const options = multiple ? [...selectedOptions, option] : option;
    setSelectedOptions([...selectedOptions, option]);
    onChange && onChange(options);
  }
const isSelected = multiple ? selectedOptions.includes(option) : selectedIndex === index;
```
But I'm not convinced with what I did,  I think there is should better way what do you think?

I would be really really happy and grateful if you could help me with it

**Reply by atomiks** [reply]
I'm not 100% sure what you are wanting to do, but I made a multi-select with groups example that handles all interactions including keyboard (without a proper component API) here: 

https://codesandbox.io/s/charming-jackson-k8f1v7?file=/src/App.tsx

Does this help?
